sway-lib-core/src/primitive_conversions.sw

```diff
@@ -248,6 +270,28 @@ impl u32 {
         }
     }
 
+    /// Extends a `u32` to a `u256`.
+    ///
+    /// # Returns
+    ///
+    /// * [u256] - The converted `u32` value.
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// fn foo() {
+    ///     let val = 2u32;
+    ///     let result = val.as_u256();
+    ///     assert(result == 0x0000000000000000000000000000000000000000000000000000000000000002u256);
+    /// }
+    /// ```
+    pub fn as_u256(self) -> u256 {
+        let input = (0, 0, 0, self);
```

Comment:
I think this is a little bit dangerous... I would convert `self` to u64 first. This can be typed as (u32, u32, u32, u32) and generate the wrong binary representation.

Same for other integer types.